packages/playground/src/components/node_selector/TfAutoNodeSelector.vue

```diff
@@ -299,14 +301,23 @@ export default {
 
     const nodeInputValidateTask = useAsync<boolean, string, [NodeInfo | undefined]>(
       async node => {
-        if (node && node?.rentContractId !== 0) {
-          const { state } = await gridStore.grid.contracts.get({
-            id: node?.rentContractId,
-          });
-          if (state.gracePeriod) {
-            return false;
+        try {
+          if (node && node.rentContractId !== 0) {
+            const { state } = await gridStore.grid.contracts.get({
+              id: node?.rentContractId,
+            });
+            if (state.gracePeriod) {
+              throw `You can't deploy on node ${node.nodeId}, its rent contract is in grace period.`;
```

Comment:
i suggest adding time out before throwing the error, for more friendly UX.

Reply:
why do we need timeout here? we check the state object only.

Reply:
https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3646#pullrequestreview-2448787416
in this video, the switching between info and error toast is to fast, if you have any other solutoin to hide the info toast then show the error will be much better imo
[Screencast from 2024-12-01 11-27-36.webm](https://github.com/user-attachments/assets/25278603-d7a4-4848-9299-9cf5b2d7d6d0)

Reply:
fixed